packages/flutter/test/material/flat_button_test.dart

```diff
@@ -333,6 +333,79 @@ void main() {
         paintsExactlyCountTimes(#clipPath, 0),
     );
   });
+
+  testWidgets('FlatButton responds to tap and onLongPress when enabled', (WidgetTester tester) async {
+
+    int pressedCount = 0;
+
+    Widget buildFrame({VoidCallback onPressed, VoidCallback onLongPress}) {
+      return Directionality(
+        textDirection: TextDirection.ltr,
+        child: FlatButton(child: const Text('button'), onPressed: onPressed, onLongPress: onLongPress),
+      );
+    }
+
+    // onPressed not null, onLongPress null.
+    await tester.pumpWidget(
+      buildFrame(onPressed: () { pressedCount += 1; }, onLongPress: null),
+    );
+    expect(tester.widget<FlatButton>(find.byType(FlatButton)).enabled, true);
+    await tester.tap(find.byType(FlatButton));
+    await tester.pumpAndSettle();
+    expect(pressedCount, 1);
+
+    // onPressed null, onLongPress not null.
+    pressedCount = 0;
+    await tester.pumpWidget(
+      buildFrame(onPressed: null, onLongPress: () { pressedCount += 1; }),
+    );
+    expect(tester.widget<FlatButton>(find.byType(FlatButton)).enabled, true);
+    await tester.longPress(find.byType(FlatButton));
+    await tester.pumpAndSettle();
+    expect(pressedCount, 1);
+
+    // onPressed null, onLongPress null.
+    pressedCount = 0;
+    await tester.pumpWidget(
+      buildFrame(onPressed: null, onLongPress: null),
+    );
+    expect(tester.widget<FlatButton>(find.byType(FlatButton)).enabled, false);
+    await tester.tap(find.byType(FlatButton));
```

Comment:
I think checking for taps and long presses when the button is disabled may not be useful. Since we're not passing a callback into `onPressed` or `onLongPress`, I wouldn't imagine `pressedCount` would increment. 

Same for the other button tests

Reply:
So.. In that situation, just keep the enabled test and remove the tap+longPress tests?

Reply:
Yes